File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Achievements/CharacterAchievementsModel.cs
using CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.MythicKeystone;
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Achievements;

public class CharacterAchievementsModel
{
    [JsonPropertyName("total_quantity")]
    public int TotalQuantity { get; set; }

    [JsonPropertyName("total_points")]
    public int TotalPoints { get; set; }

    [JsonPropertyName("achievements")]
    public CharacterAchievementModel[] Achievements { get; set; }

    [JsonPropertyName("category_progress")]
    public CharacterAchievementCategoryModel[] CategoryProgress { get; set; }

    [JsonPropertyName("recent_events")]
    public CharacterAchievementRecentEventsModel[] RecentEvents { get; set; }

    [JsonPropertyName("character")]
    public DungeonCharacterModel Character { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/achievements/CharacterAchievementsModel.ts
import type { DungeonCharacterModel } from '../mythicKeystone/DungeonCharacterModel';
import type { CharacterAchievementCategoryModel } from './CharacterAchievementCategoryModel';
import type { CharacterAchievementModel } from './CharacterAchievementModel';
import type { CharacterAchievementRecentEventsModel } from './CharacterAchievementRecentEventsModel';

export type CharacterAchievementsModel = {
    totalQuantity: number;
    totalPoints: number;
    achievements: CharacterAchievementModel[];
    categoryProgress: CharacterAchievementCategoryModel[];
    recentEvents: CharacterAchievementRecentEventsModel[];
    character: DungeonCharacterModel;
}